### ISSUE TITLE
Darwin check fails

### ISSUE THREAD

**Issue by polluks** [gwsw/less#426]
```
$ uname -a
Darwin mbp 23.0.0 Darwin Kernel Version 23.0.0: Tue Aug 22 02:13:07 PDT 2023; root:xnu-10002.1.11~16/RELEASE_ARM64_T8103 arm64
$ make check
if [ -x less ] && ./less -V | grep -q LESSTEST; then :; else /Applications/Xcode.app/Contents/Developer/usr/bin/make clean; /Applications/Xcode.app/Contents/Developer/usr/bin/make LESSTEST=1; fi
objdir=$(pwd); rflags=""; if [ -z "" ]; then rflags=-e; fi; cd ./lesstest && /Applications/Xcode.app/Contents/Developer/usr/bin/make && ./runtest $rflags -l "$objdir/less" lt
make[1]: Nothing to be done for `all'.
DIFF table-50x200 on cmd #63 (. a)
FAIL: table-50x200 (63 steps)
ERR  status 256 from /Users/polluks/g/less/lesstest/lesstest  -e -s '/Users/polluks/g/less/lesstest/lt_screen' -t '/Users/polluks/g/less/lesstest/lt/table-50x200.lt' '/Users/polluks/g/less/less'
DIFF colorbars on cmd #96 (. a)
FAIL: colorbars (96 steps)
ERR  status 256 from /Users/polluks/g/less/lesstest/lesstest  -e -s '/Users/polluks/g/less/lesstest/lt_screen' -t '/Users/polluks/g/less/lesstest/lt/colorbars.lt' '/Users/polluks/g/less/less'
RAN  13 tests with 2 errors
make: *** [check] Error 1
```
(this is not  #293)

**Comment by awmorgan** [comment]
Hey @polluks,

I was able to reproduce the issue on my end.

After running a git bisect, I identified the first bad commit as:
```
236e0419e86ec323ea6b10bbd22c67e7c6bd76bd is the first bad commit
commit 236e0419e86ec323ea6b10bbd22c67e7c6bd76bd
Author: Mark Nudelman <markn@greenwoodsoftware.com>
Date:   Mon Sep 11 12:41:54 2023 -0700

    Use simpler and more efficient algorithm to make search match
    visible when -S is in effect.

 search.c | 32 +++-----------------------------
 1 file changed, 3 insertions(+), 29 deletions(-)
```

**Comment by gwsw** [comment]
Fixed in ca3ecdb0c68854ac112fdfae54a1a11d19611ee5.
